### PR TITLE
update metrics/Dockerfile to match current binary name format

### PR DESCRIPTION
**Pull request by Andreea-Lupu**
Signed-off-by: Andreea-Lupu <andreealupu1470@yahoo.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:
After changing the name format of the binary(ex zot-$OS-$ARCH) trying to build images using metrics/Dockerfile doesn't work anymore because of name mismatch. This PR fixes this issue by updating the name of the binary to current format.

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/metrics/Dockerfile [hunk 1]
```diff
@@ -30,11 +32,13 @@ RUN echo '{\n\
 # ---
 # Stage 2: Final image with nothing but certs, binary, and default config file
 # ---
-FROM scratch AS final
+FROM gcr.io/distroless/base AS final
+ARG OS
+ARG ARCH
 COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
-COPY --from=builder /go/src/github.com/project-zot/zot/bin/zot /zot
```
Comment (adodon2go) [inline examples/metrics/Dockerfile:35]: I think what we need to do is to keep this file similar to Dockerfile-conformance from which this file was originally inspired from (maybe rename Dockerfile -> Dockerfile-conformance ?)

I'm thinking of changing this line to:
`COPY --from=builder /go/src/github.com/project-zot/zot/bin/zot-linux-amd64 /usr/bin/zot`
There ios no need for OS & ARCH args as the kind-setup.sh script was intended to be run on linux systems and was not tested on other platforms.

File: examples/metrics/Dockerfile-zxp [hunk 2]
```diff
@@ -23,9 +25,11 @@ RUN echo '{\n\
 # ---
 # Stage 2: Final image with nothing but binary and default config file
 # ---
-FROM scratch AS final
-COPY --from=builder /go/src/github.com/project-zot/zot/bin/zxp /zxp
+FROM gcr.io/distroless/base AS final
+ARG OS
+ARG ARCH
+COPY --from=builder /go/src/github.com/project-zot/zot/bin/zxp-$OS-$ARCH /usr/bin/zxp
```
Comment (adodon2go) [inline examples/metrics/Dockerfile-zxp:31]: Better to use zxp-linux-amd64 directly and get rid of OS & ARCH args

File: examples/metrics/Makefile [hunk 3]
```diff
@@ -1,8 +1,10 @@
 CONTAINER_RUNTIME := docker
+OS ?= linux
```
Comment (adodon2go) [inline examples/metrics/Makefile:2]: I don't think we really need to change this file.

File: examples/metrics/Dockerfile-zxp (hunk 2 above)
Comment (rchincha) [reply examples/metrics/Dockerfile-zxp:31]: No why? I can reuse the same Dockerfile and it gets OS and ARCH from ci/cd

Comment (adodon2go) [reply examples/metrics/Dockerfile-zxp:31]: The idea was that kind-setup.sh script was never tested on other platforms to prove it works.
If the idea was to bring a kind cluster with all the zot pods installed (+ prometheus) than we need to validate that script works on different flavours of Mac processors too.

**Pull request merged**